client/app/pages/alert/Alert.jsx

```diff
@@ -0,0 +1,405 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { head, includes, template as templateBuilder, trim } from 'lodash';
+import cx from 'classnames';
+
+import { $route } from '@/services/ng';
+import { currentUser } from '@/services/auth';
+import navigateTo from '@/services/navigateTo';
+import notification from '@/services/notification';
+import { Alert as AlertService } from '@/services/alert';
+import { Query as QueryService } from '@/services/query';
+
+import { HelpTrigger } from '@/components/HelpTrigger';
+import LoadingState from '@/components/items-list/components/LoadingState';
+import { TimeAgo } from '@/components/TimeAgo';
+
+import Form from 'antd/lib/form';
+import Button from 'antd/lib/button';
+import Icon from 'antd/lib/icon';
+import Modal from 'antd/lib/modal';
+import Input from 'antd/lib/input';
+import Dropdown from 'antd/lib/dropdown';
+import Menu from 'antd/lib/menu';
+
+import Criteria from './components/Criteria';
+import Rearm from './components/Rearm';
+import Query from './components/Query';
+import { STATE_CLASS } from '../alerts/AlertsList';
+import { routesToAngularRoutes } from '@/lib/utils';
+
+
+const defaultNameBuilder = templateBuilder('<%= query.name %>: <%= options.column %> <%= options.op %> <%= options.value %>');
+const spinnerIcon = <i className="fa fa-spinner fa-pulse m-r-5" />;
+
+function isNewAlert() {
```

Comment:
Why not create two page components -- one for new alert and one for edit page, which reuse components between them?

Having ~10 places with conditional logic based on this doesn't feel right.

Reply:
Ok, I'll try it and get back to ya.

Reply:
@arikfr it makes more sense to have New and Edit as one - as they share almost all components and behaviors, and have a different page for view mode.

Wdyt?

Reply:
I didn't realize view mode was mixed in there as well. This one should definitely be separate.

Reply:
Done in https://github.com/getredash/redash/pull/4183